### ISSUE TITLE
Where is the batch size set for the validation set during training

### ISSUE THREAD

**Issue by litengfeigithub** [ultralytics/ultralytics#16793]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Where is the batch size set for the validation set during training

### Additional

Where is the batch size set for the validation set during training

**Comment by UltralyticsAssistant** [comment]
👋 Hello @litengfeigithub, thank you for reaching out to Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for more insights and where you may find details about validation batch size settings.

If this is a custom training ❓ Question, please share additional information such as dataset image examples and training logs. Ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For more interactive support, join the Ultralytics community via [Discord](https://ultralytics.com/discord) 🎧. If you prefer in-depth discussions, [Discourse](https://community.ultralytics.com) is a great place to engage, and feel free to connect with others on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you're using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO is compatible with several environments (with all dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. For guidance, see [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Refer to [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Details in [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, confirming correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across different platforms.

This is an automated response. An Ultralytics engineer will assist soon! 😊

**Comment by Y-T-G** [comment]
It's double the training batch size.

**Comment by Y-T-G** [comment]
Double if not using OBB, in which case it's the same as training.

https://github.com/ultralytics/ultralytics/blob/8154a27e9b332dd2dac7a107003462f0e20f6179/ultralytics/engine/trainer.py#L294-L296

**Comment by litengfeigithub** [comment]
> Double if not using OBB, in which case it's the same as training.
> 
> https://github.com/ultralytics/ultralytics/blob/8154a27e9b332dd2dac7a107003462f0e20f6179/ultralytics/engine/trainer.py#L294-L296

thank you very much

**Comment by syorami** [comment]
> Double if not using OBB, in which case it's the same as training.
> 
> https://github.com/ultralytics/ultralytics/blob/8154a27e9b332dd2dac7a107003462f0e20f6179/ultralytics/engine/trainer.py#L294-L296

hi @Y-T-G does the `batch_size` here mean the batch size per rank when multi gpu training? I found that when setting bs=128 with 8gpus, the validation bs is actually 32. How can I enlarge the validation bs?

**Comment by Y-T-G** [comment]
Validation runs on a single GPU, so you will have to modify the code if you want it to be more than (128/8)*2
